### ISSUE TITLE
Rule suggestion: check qualifier for parameter-properties 

### ISSUE THREAD

**Issue by nrusitska** [palantir/tslint#4651]
### Rule Suggestion
It would be great to have rule that enforces 'this' qualified access to constructor's parameter properties in the constructor code.
basically this suggestion is for a rule that prevents the developer from an error like this:

```typescript
public constructor(private name: string) {
   name = name.replace('foo', 'bar');
}
```
While expecting that the 'name' property does not contain the string 'foo' anymore.

In this case the developer wanted to transform the value of the property 'this.name', but he did only change the value of the local variable 'name' (which is shadowed by the parameter property).
whilst the correct code would look like this:

```typescript
public constructor(private name: string) {
   this.name = this.name.replace('foo', 'bar');
}
```

**Comment by JoshuaKGoldberg** [comment]
Interesting idea. What would be expected with code that uses the modified variable after the fact?

Example:

```ts
public constructor(private name: string) {
   name = name.replace('foo', 'bar');
   console.log(name);
}
```

...should the rule only capture parameter property arguments that are modified after any code that would work on them is done?

**Comment by adidahiya** [comment]
Also note that the tsc no unused parameters check would produce an error here:

![image](https://user-images.githubusercontent.com/723999/56245721-1bc3da00-606e-11e9-89eb-128de52136e9.png)

**Comment by nrusitska** [comment]
@Josh, I’d would like the rule to capture every time the local variable is used at all, to prevent unwanted usage. 
@Adi, yeah, that’s a good start but in the scenario I’m thinking about, especially the mixed access on “name” and “this.name” is causing the unexpected behaviour.

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for new rules. See #4534. 😱

If you'd like to see this rule implemented, you have two choices:
* **Recommended:** Use [typescript-eslint](https://typescript-eslint.io) to lint your TypeScript code with ESLint ✅
* **Legacy:** Write it as a [custom TSLint rule](https://palantir.github.io/tslint/develop/custom-rules/) 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
